koku/reporting/provider/aws/models.py

```diff
@@ -283,6 +283,7 @@ class Meta:
         indexes = [models.Index(fields=["key", "enabled"], name="aws_enabled_category_key_index")]
         db_table = "reporting_awsenabledcategorykeys"
 
+    uuid = models.UUIDField(default=uuid4, editable=False, unique=True, null=False)
```

Comment:
What is the purpose of this field?

Reply:
The addition of the uuid is part of the new design for enabling & disabling keys through the new settings page APIs. Admittedly, we are changing the aws categories enablement here to be consistent with how we will be enabling tags in the future. 

Our old enabling/disabling tag workflow required you to know the tag key & the provider type of the key. However, we are now introducing uuids to these tables so that we can pass a list of uuids to the enable/disable endpoints for enablement. 

Examples of this can be found in this pull request: https://github.com/project-koku/koku/pull/4466

Reply:
I guess I still don't understand the point. The `key` is the primary key for this table, which automatically makes it unique. Where will we be getting the uuids to enable the keys? and why can we not just enable keys thru the primary key?

Reply:
I have the same question here.

Reply:
The missing piece here is that provider specific tables will be [combined into one table](https://github.com/project-koku/koku/compare/main...COST-3752-tags-settings-api?diff=split) with all keys for all providers. That introduces the potential for collisions in the `key` field.

Reply:
Sam mentioned the tag part having a collision on provider which is why a UUID was added there. The intended purpose of adding it here was to keep the API response and settings flow for enabling/disabling Cost Categories consistent with how Tags are enabled/disabled, both GET responses would include a UUID for each record that could then be used to enable/disable the relevant record. 

Its feasible to just use the key here, it's worth asking Dan if the consistency of a UUID brings him value or not as I would imagine the UI is how people will interact with their settings.

Reply:
If we're trying to make things consistent, that is fine. I think we need to update the table here such that the UUID is the PK instead of the key. It just doesn't make sense to add a random UUID if we don't use it as a PK.

Reply:
Makes the migrations a bit more complicated to use a new PK, but I finally got it figured out: https://github.com/project-koku/koku/pull/4464/commits/40e21f8e86eae3e84964786e9e188d9eae2c9d03